File: app/src/main/java/com/tsuruta/meet/recycler/EventRecyclerAdapter.java
package com.tsuruta.meet.recycler;

import android.support.v7.widget.RecyclerView;
import android.text.format.DateFormat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.tsuruta.meet.R;
import com.tsuruta.meet.fragments.EventListFragment;
import com.tsuruta.meet.objects.Event;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

public class EventRecyclerAdapter extends RecyclerView.Adapter<EventRecyclerViewHolder>
{
    private ArrayList<Event> events;
    private ArrayList<ArrayList<String>> urlList;
    private EventListFragment parent;

    public EventRecyclerAdapter(EventListFragment parent, ArrayList<Event> events, ArrayList<ArrayList<String>> urlList)
    {
        this.events = events;
        this.urlList = urlList;
        this.parent = parent;
    }

    public void updateList(ArrayList<Event> events, ArrayList<ArrayList<String>> urlList)
    {
        this.events = events;
        this.urlList = urlList;
        notifyDataSetChanged();
    }

    public void eventClicked(int position)
    {
        if(events.get(position).getHasJoined())
        {
            parent.eventClicked(position);
        }
    }

    public void joinEvent(int position)
    {
        parent.joinEvent(position);
    }

    public void acceptInvite(int position)
    {
        parent.acceptInvite(position);
    }

    public void denyInvite(int position)
    {
        parent.denyInvite(position);
    }

    @Override
    public int getItemCount()
    {
        if(events == null)
        {
            return 0;
        }
        else
        {
            return events.size();
        }
    }

    @Override
    public EventRecyclerViewHolder onCreateViewHolder(ViewGroup viewGroup, int position)
    {
        LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
        View itemView = inflater.inflate(R.layout.card_event, viewGroup, false);
        return new EventRecyclerViewHolder(this, itemView);
    }

    @Override
    public void onBindViewHolder(EventRecyclerViewHolder viewHolder, int position)
    {
        if (events.get(position) != null)
        {
            String name = events.get(position).getTitle();

            //Find creator's name based on userUid
            String creator = events.get(position).getCreatorName();

            //Calculate expiration date based on the creation date
            long timestamp = events.get(position).getTimestamp();
            String expires = getDate(timestamp);

            boolean hasJoined = events.get(position).getHasJoined();

            viewHolder.setTvEventTitle(name);
            viewHolder.setTvEventCreator(creator);
            viewHolder.setTvEventExpires(expires);
            viewHolder.setJoinVisibility(hasJoined);
            viewHolder.setAvatarUrls(urlList.get(position));
            viewHolder.setRowIndex(position);

            if(events.get(position).getInvited())
            {
                //TODO: Pass into this method the user's name.
                viewHolder.setInviter("$user_name");
            }
        }
    }

    public EventListFragment getParent()
    {
        return parent;
    }

    private String getDate(long timestamp)
    {
        Calendar cal = Calendar.getInstance(Locale.ENGLISH);
        cal.setTimeInMillis(timestamp);
        return DateFormat.format("MM-dd-yyyy", cal).toString();
    }
}

File: app/src/main/java/com/tsuruta/meet/recycler/EventRecyclerAdapter.kt
package com.tsuruta.meet.recycler

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.bumptech.glide.Glide

import com.tsuruta.meet.R
import com.tsuruta.meet.fragments.FindEventFragment
import com.tsuruta.meet.objects.Event

import java.util.ArrayList
import android.os.Build
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.Drawable
import android.graphics.Bitmap
import com.bumptech.glide.request.animation.GlideAnimation
import com.bumptech.glide.request.target.SimpleTarget



class EventRecyclerAdapter(val parent: FindEventFragment, private var events: ArrayList<Event>) : RecyclerView.Adapter<EventRecyclerViewHolder>()
{
    fun updateList(events: ArrayList<Event>)
    {
        this.events = events
        notifyDataSetChanged()
    }

    fun eventClicked(position: Int)
    {
        parent.eventClicked(position)
    }

    override fun getItemCount(): Int = events.size

    override fun onCreateViewHolder(viewGroup: ViewGroup, position: Int): EventRecyclerViewHolder
    {
        val inflater = LayoutInflater.from(viewGroup.context)
        val itemView = inflater.inflate(R.layout.card_event, viewGroup, false)
        return EventRecyclerViewHolder(this, itemView)
    }

    override fun onBindViewHolder(viewHolder: EventRecyclerViewHolder, position: Int)
    {
        viewHolder.tvTitle.text = events[position].name
        viewHolder.tvLocation.text = events[position].location
        viewHolder.tvTime.text = events[position].time
        viewHolder.tvDate.text = events[position].date
        Glide
                .with(parent.getParent())
                .load(events[position].imageUrl)
                .asBitmap()
                .centerCrop()
                .placeholder(R.drawable.com_facebook_button_like_background)
                .into(object : SimpleTarget<Bitmap>(200, 200)
        {
            override fun onResourceReady(resource: Bitmap, glideAnimation: GlideAnimation<in Bitmap>)
            {
                val drawable = BitmapDrawable(parent.resources, resource)
                viewHolder.rlEventCard.background = drawable
            }
        })
    }
}
